versioned/persist/adapter/src/test/java/org/projectnessie/versioned/persist/adapter/spi/TestKeyListBuildState.java

```diff
@@ -154,6 +155,25 @@ static Stream<Arguments> openAddressing() {
         arguments(20000, MINIMUM_BUCKET_SIZE, MINIMUM_BUCKET_SIZE * 32, 8, 79, 0.45f, 1));
   }
 
+  @Test
+  public void nextPowerOfTwoBucketCountb() {
+    // This case involves a computed shift distance of 32 on type int.  As described in JLS 11
+    // section 15.19 Shift Operators, only the five lowest-order bits of the distance are actually
+    // used for an int's shift distance, so the effective shift distance is 0.
+    assertThat(KeyListBuildState.nextPowerOfTwo(0)).isEqualTo(1);
+
+    assertThat(KeyListBuildState.nextPowerOfTwo(1)).isEqualTo(1);
+    assertThat(KeyListBuildState.nextPowerOfTwo((1 << 29) + 1)).isEqualTo(1 << 30);
+    assertThat(KeyListBuildState.nextPowerOfTwo((1 << 30) - 1)).isEqualTo(1 << 30);
+    assertThat(KeyListBuildState.nextPowerOfTwo(1 << 30)).isEqualTo(1 << 30);
+    assertThat(KeyListBuildState.nextPowerOfTwo((1 << 30) + 1)).isEqualTo(1 << 31);
+    assertThat(KeyListBuildState.nextPowerOfTwo(1 << 31)).isEqualTo(1 << 31);
+    // TODO might just want to throw an exception early in the following cases, in nextPowerOfTwo
```

Comment:
Duh, yes! Please let it either throw an exception, before it becomes a negative value.

Reply:
Sounds good.  I've set the range of valid inputs to `[0, 2^30]`, with a `checkArgument` precondition, javadoc, and test (the same new parameterized test that exercises a negative-parameter-value exception, commented on further down).